### ISSUE TITLE
Pjax and Pagination with ExpandRowColumn 

### ISSUE THREAD

**Issue by GerhardK90** [kartik-v/yii2-grid#782]
## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest `master` branch of yii2-grid.
- [x] This is not an usage question. I confirm having gone through and read the [documentation](http://demos.krajee.com/grid) and [demos](http://demos.krajee.com/grid-demo).
- [x] This is not a general programming / coding question. (Those should be directed to the [webtips Q & A forum](http://webtips.krajee.com/questions)).
- [x] I have attempted to find the simplest possible steps to reproduce the issue.
- [x] I have included a failing test as a pull request (Optional).

## Steps to reproduce the issue

1.  Create a GridView, with Pjax enabled
2.  Add an ExpandRowColumn to this GridView widget and render a detail view
3. Add a GridView to your detail view with Pjax enabled 
4. Expand a row, which got a GridView which has pagination

## Expected behavior and actual behavior

The usage of the pagination changes the data-index of the table row which holds the detail view to 'undefined' and adds a new table row with the correct data-index (e.g. 0 for the first row).
When collapsing the row the new table row remains. 

Expanding all rows and using the pagination of the detail view of a random row collapses all rows except one row. For every row that has been collapsed an empty row remains.   
![image](https://user-images.githubusercontent.com/3868214/37248457-12d14086-24d3-11e8-9bbc-5c0fbda668e6.png)

#### Browsers

- [x] Google Chrome
- [x] Mozilla Firefox
- [x] Internet Explorer
- [x] Safari

#### Operating System

- [x] Windows
- [x] Mac OS X
- [x] Linux
- [x] Mobile

#### Libraries

- jQuery version: 
- yii2-grid version: latest

## Isolating the problem

- [ ] This bug happens [on the demos page](https://demos.krajee.com/grid-demo)
- [x] The bug happens consistently across all tested browsers
- [x] This bug happens when using yii2-grid without other plugins.

A solution is to disable pjax in the main GridView (Step one in 'steps to reproduce the issue').

I might be unfamiliar with pjax but it is nowhere mentioned, that the main GridView should not be pjax if pjax is used within ExpandRowColumn.

**Comment by arjenmeijer** [comment]
Well, part of the solution is using the pagination object in the gridview.
```
    $dataProvider = new ArrayDataProvider([
        'allModels' => $model->mytabel,
        'key' => 'mykeyid',
        'pagination' => [
            'route' => 'mytabel/index',
        ]             
    ]);
```
This sets the route in the paging link to the correct controller/action.

Secondly, every gridview has to have an unique id. Add the id line
```
    echo GridView::widget([
        'id' => 'mytabelnameofsomethingshorter'.$model->nameofthekeyofthemothertable,
```
to every view. Pjax has to know which container the url is going to load in. In the latest version of Yii2 the id is added automatically.

Third, select only the pagelinks in a subview with a dom selector of an event. I did not yet find a solution that works always, for the next and previous links also. I would like to add a class to every subview pagelink, but I do not yet know the way to do it nicely. I have to add a class with linkpager, but I don't know how to do it in a default gridview.

`    $(document).on('click', 'a[data-page]', function(event) {`

This selects all page links, not only in the subview and is thus incorrect.

Fourth, attach a pjax event that loads the links in the correct container. I do not have the complete correct code yet. 
```
        event.stopPropagation();
        event.preventDefault();
        let container = $(this).closest('[data-pjax-container]');
        let containerSelector = '#' + container[0].id;
```

And than it works, I hope. Who can make it completely working?

P.S. According to some sites it should work out of the box, but I doubt it.

**Comment by ered15** [comment]
same issue here, any one know a fix around? liked stated by @GerhardK90 solution is to disable pjax in the main GridView and it starts working, but what if we want to use PJAX in main GridView and inside ExpandRowColumn? I also noticed the issue happens when column is expanded so another solution I tested was collapse all before triggering any pjax event inside ExpandRowColumn but it's just to many steps.
